pkg/resources/helpers_test.go

```diff
@@ -258,3 +259,147 @@ func TestListDiff(t *testing.T) {
 		})
 	}
 }
+
+func Test_DataTypeIssue3007DiffSuppressFunc(t *testing.T) {
+	testCases := []struct {
+		name     string
+		old      string
+		new      string
+		expected bool
+	}{
+		{
+			name:     "different data type",
+			old:      string(sdk.DataTypeVARCHAR),
+			new:      string(sdk.DataTypeNumber),
+			expected: false,
+		},
+		{
+			name:     "same number data type without arguments",
+			old:      string(sdk.DataTypeNumber),
+			new:      string(sdk.DataTypeNumber),
```

Comment:
I think this case (old==new) is handled by Terraform and diff suppress isn't executed.

Reply:
You are most probably right but it does not hurt us to handle this case properly too, right?